src/test/test.sh

```diff
@@ -0,0 +1,12 @@
+g++ -g -std=c++17 -Wall -I catch.hpp -I test_bed.hpp -I ../ \
```

Comment:
This seems to be very MacOS specific. Of course, we are mostly using MacOS but I think it would be better to have a proper CMake configuration.

Reply:
Because it should work for us, edge cases running Linux are welcome to contribute with a PR :).

Reply:
Because it should worked me, edge cases are welcome to contribute with a PR :).